FreeTubeApp/FreeTube
# Channels: Add support for release and podcast tab (Local API)

# Channels: Add support for release and podcast tab (Local API)

## Pull Request Type

## Related issue
closes #3763

## Description
Adds Podcasts and Releases tabs to channels (local api only). 
This PR also hides missing tabs from the UI.

## Screenshots 
![image](https://github.com/FreeTubeApp/FreeTube/assets/78101139/657beeea-7323-46be-a581-ba604d341caf)

![image](https://github.com/FreeTubeApp/FreeTube/assets/78101139/3964eb3b-4d47-4cd4-8734-2fdac21f4a6b)

## Testing 
Channel with podcasts: https://www.youtube.com/channel/UCGk1LitxAZVnqQn0_nt5qxw
- Go to the channel
- See the podcast tab
- Click on podcast tab
- Go to a different page
- Enter https://www.youtube.com/channel/UCGk1LitxAZVnqQn0_nt5qxw/podcasts in the search box
- see the podcast tab
- Go to settings > distraction free  and toggle Hide Channel Podcasts
- Go to https://www.youtube.com/channel/UCGk1LitxAZVnqQn0_nt5qxw/podcasts
- See video tab instead


Channel with releases: https://www.youtube.com/channel/UC5CwaMl1eIgY8h02uZw7u8A
- Go to the channel
- See the release tab
- Click on release tab
- Go to a different page
- Enter https://www.youtube.com/channel/UCGk1LitxAZVnqQn0_nt5qxw/releases in the search box
- see the release tab
- Go to settings > distraction free  and toggle Hide Channel Releases
- Go to https://www.youtube.com/channel/UCGk1LitxAZVnqQn0_nt5qxw/releases
- See video tab instead


Other test (trying to access a tab that doesn't exist)
- Go to https://www.youtube.com/channel/UCGk1LitxAZVnqQn0_nt5qxw/releases
- See video tab instead

## Desktop

- **OS:** Windows
- **OS Version:** 11
- **FreeTube version:** 0.18.0

## Additional context
IV support will be added in a future PR once https://github.com/iv-org/invidious/pull/3980 is merged
